### ISSUE TITLE
Whitespaces in blocks

### ISSUE THREAD

**Issue by SamyPesse** [mozilla/nunjucks#351]
I'm currently working on switching from **swig** to **nunjucks** for [gitbook.com](https://www.gitbook.com).

Here is a part of one of the templates, that is no longer working correctly:

```
<h3 class="title"><a href="{{ post.link }}">{{ post.title }}</a></h3>
<p class="description">{{ post.description }}</p>
```

But if I change the template to remove whitespaces in blocks: `{{post.title}}`, the template is working correctly.

This problem is not happening when using filters (for example `{{ post.published|fromNow }}` is working fine).

I have the same issue with macros (not all the time), for example `{{ testMacro() }}` throws an error that the macro " testMacro" doesn't exist (with a space in the error), and `{{testMacro()}}` is working fine.

Is it a bug or is there a whitespace policy that I missed?

**Comment by jlongster** [comment]
That is very strange, nothing should be enforced regarding whitespace within tags, it should be ignore. Are you running this code inside a custom extension? I wonder if that's messing with anything.

Can you paste the error the `{{ post.title }}` shows? I'll be happy to help debug it. Even better, if you have a test code online somewhere, but I know that's more work.

**Comment by SamyPesse** [comment]
`{{ post.title }}` is not throwing an error, it just show nothing (but `{{post.title}}` works). An error is throwned only by macro. I think it's the same problem, nunjucks is adding the whitespaces to the name that is trying to resolve.

Here a part of my template:

```
<div class="post">
    <h3 class="title"><a href="{{ post.link }}">{{ post.title }}</a></h3>
    <p class="description">{{ post.description }}</p>
    <ul class="blog-metadata">
        <li class="date"><i class="octicon octicon-calendar"></i> {{ post.published|fromNow }}</li>
        {% if post.author %}
        <li class="author">
            <img src="{{ post.author.email|gravatar }}" />
            <a href="{{ post.author.uri }}">{{ post.author.name }}</a>
        </li>
        {% endif %}
        <li class="category"><i class="octicon octicon-file-directory"></i> <a href="/blog/{{ post.category }}">{{ post.category }}</a></li>
    </ul>
</div>
```

Everything is working fine except: `{{ post.link }}`, `{{ post.title }}` and `{{ post.description }}`. The others (`{{ post.author.name }}`) are working correctly.

I'm using the my branch (https://github.com/mozilla/nunjucks/pull/349) and two extensions (I tested without the extensions and it's working). I can't test without using my PR.

**Comment by SamyPesse** [comment]
Here is an error thrown when using `{{ profile.link() }}`.

![screen shot 2015-01-12 at 23 37 35](https://cloud.githubusercontent.com/assets/845425/5712525/76bfd7ae-9ab3-11e4-8c32-2c8d16b01b7a.png)

**Comment by SamyPesse** [comment]
The error is thrown by the second line (not the first):

```
<a href="{{ profile.link("settings") }}" class="list-group-item {% if req.path == profile.link("settings") %}active{% endif %}">Profile</a>
<a href="{{ profile.link("settings/security") }}" class="list-group-item {% if req.path == profile.link("settings/security") %}active{% endif %}">Security</a>
```

This is very strange, because if I remove the second line (or replace by `{{profile.link("settings/security")}}`), it's working and there is no error...

**Comment by jlongster** [comment]
There's something off here, and if you could, it would be awesome to be a full reproducible test case online that I can run. I suspect that something it messing with the parser, and I don't know if it's an extension or what. The fastest way to fix though would be if you could publish a page online that prints the error outside of your setup, if possible...

**Comment by SamyPesse** [comment]
I'm going to try to reproduce this bug in a script and send you this code.

**Comment by jlongster** [comment]
Wonderful!

**Comment by SamyPesse** [comment]
I made a simple code to reproduce this code, but I only succeed to reproduce it using templates from my application. Can I send you a zip by email instead of posting it on github?

**Comment by SamyPesse** [comment]
If you don't want to share your email here, just ping by email me at samypesse@gmail.com

**Comment by SamyPesse** [comment]
I fix this error locally and submit a PR (#353).

**Comment by garygreen** [comment]
Closing due to merged PR.
